3-networks/modules/base_shared_vpc/main.tf

```
@@ -94,12 +130,13 @@ resource "google_service_networking_connection" "private_vpc_connection" {
 ************************************/
 
 module "region1_router1" {
-  source  = "terraform-google-modules/cloud-router/google"
-  version = "~> 0.3.0"
-  name    = "cr-${local.vpc_name}-${var.default_region1}-cr1"
-  project = var.project_id
-  network = module.main.network_name
-  region  = var.default_region1
+  source   = "terraform-google-modules/cloud-router/google"
+  version  = "~> 0.3.0"
+  for_each = var.mode == null || var.mode == "hub" ? toset(["yes"]) : toset([])
```

nit: IIUC we want this if in default mode or hub mode, so this maybe cleaner?
```suggestion
    for_each = var.mode != "spoke" ? toset(["yes"]) : toset([])
```

---

good catch, will update